pkg/runctl/run.go

```diff
@@ -62,6 +62,34 @@ func (r *tektonRun) IsFinished() (bool, steward.Result) {
 	return true, steward.ResultErrorInfra
 }
 
+// GetMessage returns the termination message
+func (r *tektonRun) GetMessage() string {
+	var msg string
+
+	containerInfo := r.GetContainerInfo()
+	if containerInfo != nil && containerInfo.Terminated != nil {
+		msg = containerInfo.Terminated.Message
+	}
+	if msg == "" {
+		cond := r.getSucceededCondition()
+		if cond != nil {
+			return cond.Message
+		}
+	} else {
+		allMessages, err := termination.ParseMessage(msg)
+		if err == nil {
+			for _, singleMessage := range allMessages {
+				if singleMessage.Key == jfrResultKey {
+					return singleMessage.Value
+				}
+			}
+		} else {
+			return msg
+		}
```

Comment:
Put error handling code first and save the additional indentation for the 'normal' flow:
```suggestion
		if err != nil {
			return msg
		}
		for _, singleMessage := range allMessages {
			if singleMessage.Key == jfrResultKey {
				return singleMessage.Value
			}
		}
```